### PR TITLE
Add helper manual (-h --help) option

### PR DESCRIPTION
**Pull request by imperivox**
This should work as a helper manual.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -3,9 +3,18 @@ use std::env;
 pub struct Config {
     pub file_path: String,
     pub little_endianess: bool,
+    pub show_help: bool,
```
Comment (arthurbacci) [inline src/config.rs:6]: Not the most idiomatic way to do this.

File: src/config.rs [hunk 2]
```diff
@@ -3,9 +3,18 @@ use std::env;
...
 }
 impl Config {
     pub fn new(args: &[String]) -> Result<Config, &'static str> {
+        if args.contains(&"-h".to_string()) || args.contains(&"--help".to_string()) {
```
Comment (arthurbacci) [inline src/config.rs:10]: Fine, but I think this may be better treated as an error.

File: src/main.rs [hunk 3]
```diff
@@ -2,12 +2,56 @@ use byteblitz::config::Config;
 use std::env;
 use std::process;
 
+fn print_help_manual() {
+    let help_text = format!(
```
Comment (arthurbacci) [inline src/main.rs:6]: Good, thank you a lot for writing all this

File: src/main.rs [hunk 4]
```diff
@@ -2,12 +2,56 @@ use byteblitz::config::Config;
...
+    println!("{}", help_text);
+}
+
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let args: Vec<String> = env::args().collect();
     let config = Config::new(&args).unwrap_or_else(|err| {
         println!("Problem parsing arguments: {err}");
         process::exit(1);
     });
+    if config.show_help {
```
Comment (arthurbacci) [inline src/main.rs:50]: Good

**Review by arthurbacci — CHANGES_REQUESTED** [review]
Return an error instead of making changes to the Config structure.

File: src/config.rs (hunk 1 above)
Comment (arthurbacci) [reply src/config.rs]: Fixed

File: src/config.rs (hunk 2 above)
Comment (arthurbacci) [reply src/config.rs:9]: Done

**Comment by arthurbacci** [comment]
> Return an error instead of making changes to the Config structure.

Done

**Review by arthurbacci — DISMISSED** [review]

**Comment by noahra** [comment]
LGTM 🍰

**Review by noahra — APPROVED** [review]

**Pull request merged**